### PR TITLE
OKTA-225389 Create TOTP sample application for iOS

### PR DESCRIPTION
**Pull request by anastasiiaiurok-okta**

### CODE REVIEW

File: browser-sign-in/OktaBrowserSignIn/Okta.plist [hunk 1]
```diff
@@ -5,12 +5,12 @@
 	<key>scopes</key>
 	<string>openid profile offline_access</string>
 	<key>redirectUri</key>
-	<string>{redirectUri}</string>
+	<string>com.ios-okta-sdk-redirect-test-app:/callback</string>
```
Comment (IldarAbdullin-okta) [inline browser-sign-in/OktaBrowserSignIn/Okta.plist]: reset Okta.plist?

File: README.md [hunk 2]
```diff
@@ -8,4 +8,4 @@ Please find the sample that fits your use-case from the table below.
 |--------|-------------|----------|
 | [Browser Sign In](/browser-sign-in) | An application that integrates [OktaAuth SDK](https://github.com/okta/okta-sdk-appauth-ios) to perform authorization. | Authorization performed by redirecting from native app to browser. |
 | [Custom Sign In](/custom-sign-in) | An application that integrates [OktaAuth Swift SDK](https://github.com/okta/okta-auth-swift) to perform authorization. | An application that adopts native authorization to take control over authorization flow and/or provide custom UI. | 
-
+| [TOTP Generator](/totp-app) | A sample application that generates TOTP tokens. | An application that shows how developer can build their own Google Authenticator clone for their brand. |
```
Comment (IldarAbdullin-okta) [inline README.md]: I think we have to remove mentioning of `Google Authenticator` in readme. Probably better to mention `Okta Verify`?

@nbarbettini , thoughts?

File: totp-app/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Then install dependencies:
+
+```bash
+pod install
+```
+
+Now you can build and run the application.
+
+Clicking the **Add** button will allow you to setup TOTP generator for your account. You can do this either by scanning QR image or by enetryng seret key manually. Once you do this, TOTP will be refreshed automatically.
```
Comment (IldarAbdullin-okta) [inline totp-app/README.md]: enetryng -> entering

File: totp-app/TOTP App/ManualCodeEntryController.swift [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    
+    var completion: ((Token) -> Void)?
+    
+    static func create() -> ManualCodeEntryController {
+        return UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "ManualCodeEntryController")
+            as! ManualCodeEntryController
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        self.viewDidAppear(animated)
```
Comment (IldarAbdullin-okta) [inline totp-app/TOTP App/ManualCodeEntryController.swift]: viewWillAppear

File: totp-app/TOTP App/ManualCodeEntryController.swift [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        
+        let item = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(create))
```
Comment (IldarAbdullin-okta) [inline totp-app/TOTP App/ManualCodeEntryController.swift]: probably better to move this code to `viewDidAppear `

File: totp-app/TOTP App/QRScannerController.swift [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    
+    private func setupCaptureDeviceInput() {
+        let session = AVCaptureDevice.DiscoverySession(
+            deviceTypes: [.builtInDualCamera],
+            mediaType: .video,
+            position: .back
+        )
+        
+        guard let captureDevice = session.devices.first else {
+            return
```
Comment (IldarAbdullin-okta) [inline totp-app/TOTP App/QRScannerController.swift]: Any error message?

File: totp-app/TOTP App/QRScannerController.swift [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        }
+        
+        do {
+            let input = try AVCaptureDeviceInput(device: captureDevice)
+            captureSession.addInput(input)
+        } catch {
+            print(error)
```
Comment (IldarAbdullin-okta) [inline totp-app/TOTP App/QRScannerController.swift]: Alert?

File: totp-app/TOTP App/ViewController.swift [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        } catch {
+            print("Failed to store token: \(error)")
+        }
+    }
+    
+    private func deleteToken(at index: Int) {
+        let persistentToken = persistentTokens[index]
+        do {
+            try Keychain.sharedInstance.delete(persistentToken)
+            persistentTokens.remove(at: index)
```
Comment (IldarAbdullin-okta) [inline totp-app/TOTP App/ViewController.swift]: Update `tableView `?

File: totp-app/TOTP App/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import UIKit
+import OneTimePassword
+
+class ViewController: UIViewController {
```
Comment (IldarAbdullin-okta) [inline totp-app/TOTP App/ViewController.swift]: Please create name for the `ViewController `

Comment (anastasiiaiurok-okta) [reply totp-app/TOTP App/ViewController.swift]: Done.

File: totp-app/TOTP App/ManualCodeEntryController.swift (hunk 5 above)
Comment (IldarAbdullin-okta) [reply totp-app/TOTP App/ManualCodeEntryController.swift]: Sorry for confusion, I wanted to say `viewDidLoad` .  😭

**Pull request merged**